cors/304.htm

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>CORS - 304 Responses</title>
+<meta name=author title="Mark Nottingham" href="mailto:mnot@mnot.net">
+
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=support.js?pipe=sub></script>
+
+<h1>CORS - 304 Responses</h1>
+<div id=log></div>
+<script>
+
+
+/*
+ * 304 Responses
+ */
+
+// A header used to correlate requests and responses
+var state_header = "content-language"
+
+function req(url, id, t, ready) {
+  console.log('starting request ' + id)
+  var client = new XMLHttpRequest()
+  client.error_caught = false
+  try {
+    client.open('GET', url, true)
+    client.setRequestHeader(state_header, id)
+    client.send()
+  } catch (e) {
+    client.error_caught = e
+  }
+  client.onreadystatechange = function() {
+    if (client.readyState == client.DONE) {
+      console.log('finished request ' + id)
+      t.step(function() {
+        assert_true(client.status != 299, "req " + id + " server says: " + client.responseText)
+      })
+      ready(client)
+    }
+  }
+  return client
+}
+
+function two_reqs(id1, id2, should_have_same_body, t, done) {
+  var rand = Date.now()
+  var url = CROSSDOMAIN + 'resources/304.py?id=' + id1 + '&r=%s' + rand
+
+  var client1 = req(url, id1, t, function(client1) {
+    t.step(function() {
+      assert_equals(client1.response, "Success", "didn't get successful 1st response;")
+      assert_equals(client1.getResponseHeader(state_header), id1, "1st response didn't come from server;")
+    })
+
+    setTimeout(function() {
```

Comment:
Maybe add a comment to that effect? In particular it seems you want to avoid doing the second request while the first request's response might still be written to the cache. In that case it still seems better to start the timer after you received the response from the first request, no?